test/unit/document/document_test.ts

```diff
@@ -75,6 +75,24 @@ describe('DocumentReplica', function () {
     assert.equal('{"data":{"":null,"null":null}}', doc.toSortedJSON());
   });
 
+  it('delete elements of array test', function () {
+    const doc = DocumentReplica.create('test-col', 'test-doc');
+    doc.update((root) => {
+      root.data = [0, 1, 2];
+    });
+    assert.equal('{"data":[0,1,2]}', doc.toSortedJSON());
+
+    doc.update((root) => {
+      delete root.data[0];
+    });
+    assert.equal('{"data":[1,2]}', doc.toSortedJSON());
+
+    doc.update((root) => {
+      delete root.data[1];
+    });
+    assert.equal('{"data":[1]}', doc.toSortedJSON());
+  });
```

Comment:
How about adding this part too..?

```suggestion
    doc.update((root) => {
      delete root.data[0];
    });
    assert.equal('{"data":[]}', doc.toSortedJSON());
  });
```